pkgs/development/libraries/blkhash/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchFromGitLab
+, meson
+, ninja
+, pkg-config
+, cmake
+, openssl
+, libnbd
+, asciidoc
+, python3Packages
+}:
+let
+  unity = fetchFromGitHub {
+    owner = "ThrowTheSwitch";
+    repo = "Unity";
+    rev = "5204c1bacf37e0c38211d2fbb6d9796e410223f8";
+    sha256 = "i4uQOG/Q86AaQINJoC/65QA/ZEUOXzaVceR2inw32yU=";
+  };
+in
+stdenv.mkDerivation rec {
+  version = "0.6.1";
+  pname = "blkhash";
+
+  src = fetchFromGitLab {
+    owner = "nirs";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "+7SIZdvkdMYVgeklbqU0MHKc1VJnk9opqhsCmkg6bcI=";
+  };
+
+  postUnpack = ''(
+    cd "$sourceRoot/subprojects"
+    cp -R --no-preserve=mode,ownership ${unity} unity
+  )'';
```

Comment:
If we fetch the source, the normal build flow documented in the project README.md
already handles the subprojects.

Reply:
Unfortunately, that's not a case here, as nix builds packages in offline sandbox to improve reproducibility.
Fetching subprojects manually gives control over the version of dependencies, so nix is able to build the same package
with the same checksum independently from the changes on Unity repo.
I've tried to check how other packages using meson resolved that issue, and didn't find any better way to do that, but if anyone
has better solution please let me know.

Reply:
Adding `fetchSubmodules = true` to fetchFromGitLab will do the trick for you.